Mathlib/NumberTheory/NumberField/InfiniteAdeleRing.lean

```diff
@@ -0,0 +1,69 @@
+/-
+Copyright (c) 2024 Salvatore Mercuri, María Inés de Frutos-Fernández. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Salvatore Mercuri, María Inés de Frutos-Fernández
```

Comment:
BTW is this PR a collaboration with María? You may add "Co-authored-by" in the PR message.

Reply:
This is not a direct collaboration as such, but María Inés originally formalised the infinite adele ring with a different approach a couple of years ago before we had the embeddings API, and has helped with advice on my current approach which is why I put the file as co-authored, but the code in this PR is all my own. Not sure what the norm for attribution in this case would be!

Reply:
Fair enough. I thought María Inés had done this a while ago and was about to ask you to collaborate with her until I saw this.